public_html/wp-content/mu-plugins/blocks/view/sessions.php

```diff
@@ -0,0 +1,126 @@
+<?php
+namespace WordCamp\Blocks\Sessions;
+defined( 'WPINC' ) || die();
+
+use function WordCamp\Blocks\Shared\{ get_all_the_content, array_to_human_readable_list };
+use function WordCamp\Blocks\Shared\Components\{ render_featured_image };
+
+/** @var array  $attributes */
+/** @var \WP_Post  $session */
+/** @var array  $speakers */
+/** @var string $container_classes */
+
+setup_postdata( $session );
+
+?>
+
+<div class="wordcamp-block-post-list-item wordcamp-session wordcamp-session-<?php echo sanitize_html_class( $session->post_name ); ?> wordcamp-clearfix">
+	<h3 class="wordcamp-item-title wordcamp-session-title">
+		<a href="<?php echo esc_url( get_permalink( $session ) ); ?>">
+			<?php echo get_the_title( $session ); // phpcs:ignore WordPress.Security.EscapeOutput ?>
```

Comment:
In this particular situation, rather than ignoring this, I think we should add `wp_kses_post()` or similar. 

Related: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1538

----

In general, I think adding `phpcs:ignore` comments to the codebase should be rare. Seeing a lot of these comments in the codebase is a "smell" that our config file needs to be improved, and it has the effect of making the code harder to read.

I think we should only really use them in situations like [overriding `$post` in `get_contact_form_shortcode()`](https://github.com/WordPress/wordcamp.org/pull/34/files#diff-eed50fd407482588ef91f414ac7259fcR300), where it's necessary because of a true exception to the rule.

I think most of the time these comments tend to be used because a particular rule has false positives. In some cases, we can tweak the options in `phpcs.xml.dist` to allow the desired formatting. If that's not possible, though, then I'd rather see us just `<exclude>` the rule from the config file and rely on our own judgement to keep the code formatted properly, rather than litter the codebase with metadata that is essentially irrelevant at that layer.

@vedanshujain , @coreymckrill

Reply:
I agree that we should minimize exceptions. In this particular case, I assumed that since `the_title` doesn't trigger the sniff, `get_the_title` shouldn't either. But it looks like that's not [the](https://developer.wordpress.org/reference/functions/get_the_title/#comment-2150) [case](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1538).

So, I agree, `wp_kses_post` or `wp_kses_data` would be appropriate here.

This makes me wonder when it would ever make sense to use `the_title`, since it echoes the output before it can be modified further. Unless you added an escaping function to the `the_title` filter hook...

Reply:
Fixed.